enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestionDefaultView/InteractiveQuestionDefaultView.tsx

```diff
@@ -176,21 +175,61 @@ export const InteractiveQuestionDefaultView = ({
         </Box>
       </Box>
       {/* Refer to the SaveQuestionProvider for context on why we have to do it like this */}
-      {isSaveEnabled && isSaveModalOpen && question && (
-        <SaveQuestionModal
-          question={question}
-          originalQuestion={originalQuestion ?? null}
-          opened
-          closeOnSuccess
-          onClose={closeSaveModal}
-          onCreate={onCreate}
-          onSave={async question => {
-            await onSave(question);
-            closeSaveModal();
-          }}
-          targetCollection={targetCollection}
-        />
-      )}
+      <DefaultViewSaveModal isOpen={isSaveModalOpen} close={closeSaveModal} />
     </FlexibleSizeComponent>
   );
 };
+
+const DefaultViewSaveModal = ({
+  isOpen,
+  close,
+}: {
+  isOpen: boolean;
+  close: () => void;
+}) => {
+  const {
+    question,
+    originalQuestion,
+    onCreate,
+    onSave,
+    isSaveEnabled,
+    targetCollection,
+  } = useInteractiveQuestionContext();
+
+  const { id, isLoading } = useValidatedEntityId({
+    type: "collection",
+    id: targetCollection,
+  });
+
+  const finalId = id ?? targetCollection;
+
+  return (
+    isSaveEnabled &&
+    isOpen &&
+    question &&
+    !isLoading && (
```

Comment:
or you could invert this condition and early return `null` like you did in `enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/SdkSaveQuestionForm.tsx`

Having some space might make it easier to read.